webClient/src/app/app.component.ts

```diff
@@ -160,26 +160,28 @@ export class AppComponent implements AfterViewInit {
     const terminalParentElement = this.terminalParentElementRef.nativeElement;
     this.terminal = new Terminal(terminalElement, terminalParentElement, this.http, this.pluginDefinition, this.log);
     this.viewportEvents.resized.subscribe(() => this.terminal.performResize());
-    this.terminal.contextMenuEmitter.subscribe( (info) => {
-      let screenContext:any = info.screenContext;
-      screenContext["sourcePluginID"] = this.pluginDefinition.getBasePlugin().getIdentifier();
-      log.info("app.comp subcribe lambda, dispatcher="+dispatcher);
-      let recognizers:any[] = dispatcher.getRecognizers(screenContext);
-      log.info("recoginzers "+recognizers);
-      let menuItems:ContextMenuItem[] = [];
-      for (let recognizer of recognizers){
-        let action = dispatcher.getAction(recognizer);
-        log.debug("JOE:recognizer="+JSON.stringify(recognizer)+" action="+action);
-        if (action){
-          let menuCallback = () => {
-            dispatcher.invokeAction(action,info.screenContext);
+    if (this.windowActions) {
+      this.terminal.contextMenuEmitter.subscribe( (info) => {
+        let screenContext:any = info.screenContext;
+        screenContext["sourcePluginID"] = this.pluginDefinition.getBasePlugin().getIdentifier();
+        log.info("app.comp subcribe lambda, dispatcher="+dispatcher);
+        let recognizers:any[] = dispatcher.getRecognizers(screenContext);
+        log.info("recoginzers "+recognizers);
+        let menuItems:ContextMenuItem[] = [];
+        for (let recognizer of recognizers){
+          let action = dispatcher.getAction(recognizer);
+          log.debug("JOE:recognizer="+JSON.stringify(recognizer)+" action="+action);
+          if (action){
+            let menuCallback = () => {
+              dispatcher.invokeAction(action,info.screenContext);
+            }
+            // menu items can also have children
+            menuItems.push({text: action.getDefaultName(), action: menuCallback});
           }
-          // menu items can also have children
-          menuItems.push({text: action.getDefaultName(), action: menuCallback});
         }
-      }
-      this.windowActions.spawnContextMenu(info.x, info.y, menuItems);
-    });
+        this.windowActions.spawnContextMenu(info.x, info.y, menuItems);
```

Comment:
In the current state of the framework, context menu is tied to windowactions.
However, if you're doing app embedding, I'm thinking its possible to have the ability to spawn context menu on a viewport, rather than at the window level.